Add onedata templates

bwalkowi opened the pull request:
This PR introduces BYOS templates for Onedata as a file source and object store.

NOTE: we have unified naming conventions of variables which includes breaking changes (snake_case replaced camelCase). Probably, it will require config adjustments during Galaxy upgrade. Let us know if this is a problem.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

guerler committed:
Remove masthead icons

guerler committed:
Adjust masthead size

dannon committed:
Adjust SECURITY.md to reflect that we call yearly releases LTS, actual practice is the same.

guerler committed:
Remove masthead dropdown handler, only allow regular tabs

guerler committed:
Remove unused imports from masthead item component

guerler committed:
Start splitting object property into individual attributes in masthead

guerler committed:
Remove interactive tools from masthead, since already available in the activity bar

guerler committed:
Remove pseudo class handling, will use id instead

guerler committed:
Move away from json blob configuration for masthead options

guerler committed:
Decouple menu.js configuration module

guerler committed:
Delete menu.js

guerler committed:
Fix icon handling

guerler committed:
Remove unused props from masthead

guerler committed:
Fix window manager handling in masthead

guerler committed:
Reorder masthead tabs

guerler committed:
Fix icon handling, avoid reload when returning to home

guerler committed:
Fix toggle state, adjust icon styles

guerler committed:
Add configuration conditions

guerler committed:
Show navbar while loading config, reduce margin

guerler committed:
Move quota margin to navbar, adjust alignment

guerler committed:
Fix tooltips in masthead

guerler committed:
Adjust jest tests

mvdbeek committed:
Raise ``ParameterValueError`` if input collection elements deleted

This is a prerequisite to fixing https://github.com/galaxyproject/galaxy/issues/18006

guerler committed:
Preliminary adjustment of workflow run selenium test

guerler committed:
Remove additional login confirmation tests for now

guerler committed:
More selenium adjustments, using the activity bar to navigate instead of the masthead

guerler committed:
Fix tool exection in selenium tests

guerler committed:
Fix client formatting

guerler committed:
Extend list of visible activity items to anon user

guerler committed:
Fix workflow bookmarking test

guerler committed:
Fix workflow invocations selenium test

guerler committed:
Fix tour test

guerler committed:
Remove legacy navigation selectors

guerler committed:
Add help and community links to about page

guerler committed:
Adjust library identifier to align with selenium tests

guerler committed:
Fix visualization activity, adjust tour

guerler committed:
Add dropdown for user tab in masthead

guerler committed:
Remove unused popover plugin from masthead components

guerler committed:
Move activity toggle icon to the left, treat like favorites selection

guerler committed:
Reframe activity settings as list of all activities

guerler committed:
Only display optional activities when user clicks on more

guerler committed:
Fix regular icon state in activity list

guerler committed:
Adjust selenium tests with regard to updated user dropdown

guerler committed:
Adjust jest test with regard to changes in activity panel

guerler committed:
Add username to user tab in masthead

guerler committed:
Adjust quota meter, consistently show progress bar, adjust tooltip

guerler committed:
Add loading spinner to masthead

guerler committed:
Adjust quota meter test case

guerler committed:
Adjust tool running selenium test case in the context of object store

guerler committed:
Convert quota meter to composition api and typescript

guerler committed:
Adjust quota meter jest test

guerler committed:
Add total quota to quota meter

guerler committed:
Fix typescript error in jest test

guerler committed:
Add prop for library metadata, possible reactivity issue?

guerler committed:
Make masthead tooltips noninteractive

TomHarrop committed:
add pod5

(cherry picked from commit aa2b3813dd0a57023538865525dd8c951477023d)

TomHarrop committed:
reviews

(cherry picked from commit 385e7ab1e1f35c596b79efa7ddfffe0d4da753e4)

bwalkowi committed:
Add onedata file source template

bwalkowi committed:
Add onedata object store template

nsoranzo committed:
Prepare for NumPy 2.0

TomHarrop committed:
tidy linting

(cherry picked from commit 46aeb769933dc5c98780a45de4766ce7c43a80d6)

mvdbeek committed:
Replace busybox:ubuntu-14.04 image with busybox:1.36.1-glibc

dannon committed:
Pinia plugin fixes when using component mounting utilities.  This has
changed multiple times in descendant branches so this is a flattened,
cleaned up version for application against 24.0.

galaxybot committed:
Update Python dependencies

by running `make update-dependencies`.

nsoranzo committed:
Fix use of deprecated ``gxformat2._yaml`` module

Fix the following warning:

```
lib/galaxy_test/base/populators.py:79: DeprecationWarning: Importing gxformat2._yaml is deprecated, use gxformat2.yaml instead
    from gxformat2._yaml import ordered_load
```

nsoranzo committed:
Fix use of deprecated ``@abc.abstractproperty`` decorator

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,30 @@
+
+    Onedata is a global data management system providing a unified data access
+    interface across different storage providers. More information on Onedata
+    can be found in [Onedata's website](https://onedata.org/#/home).
```
on Onedata's website

lopiola commented on lib/galaxy/objectstore/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,46 @@
+
+    Onedata is a global data management system providing a unified data access
+    interface across different storage providers. More information on Onedata
+    can be found in [Onedata's website](https://onedata.org/#/home).
```
on Onedata's website

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,30 @@
+- id: onedata
+  version: 0
+  name: Onedata Storage
```
potencjlanie do dogrania, chciałbym to zobaczyć w szerszym kontekście

lopiola replied:
rozważam tutaj "Onedata virtual filesystem", chociaż to co jest teraz nie jest jakieś złe - co myślisz?

takie samo dajmy w object storze

bwalkowi replied:
nie mam preferencji - storage może być bardziej explicite dla użytkowników sobie to potem wyklikujących, ale w sumie to chyba ogarną co znaczy `virtual filesystem`

lopiola replied:
zostawmy jak jest

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,30 @@
+  version: 0
+  name: Onedata Storage
+  description: |
+    This template allows connecting to Onedata servers.
```
chyba warto rozszerzyć o to że to dla files source i że do wgrywania danych z onedata (można wspomnieć że readonly)

lopiola replied:
proponuję

If you have an account in Onedata, you can use such a file source to upload your data directly from Onedata to Galaxy. You can also perform a "remote export" of a dataset from Galaxy to Onedata. The minimal supported Onezone version is 21.02.4.

(jeśli ten tool się nie nazywa "remote export" to trezba zaktualizować)

lopiola replied:
EDIT dodałem wymaganie wersji

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,30 @@
+      default: False
+  secrets:
+    access_token:
+      label: REST Access Token
```
chyba bym wywalił stąd REST bo on nie musi być tylko restowy (może być szerszy), a w opisie jest dospecyfikowane

lopiola commented on lib/galaxy/objectstore/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,46 @@
+- id: onedata
```
tu bliźniacze uwagi co w drugim template

lopiola commented on lib/galaxy/objectstore/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,46 @@
+    space_name:
+      label: Space Name
+      type: string
+      help: The name of the Onedata space where the Galaxy data will be stored.
```
teraz pomyślałem że może ktoś mieć kolizje nazw, więc do rozważenia jak to rozegrać, pomysły:
- pozwolić podawać tutaj ID
- pozwolić podawać tutaj FQN (pewnie to zadziała od kopa? trzeba by opisać w helpie że tak można)

lopiola commented on lib/galaxy/objectstore/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,46 @@
+      label: Galaxy root directory
+      type: string
+      help: |
+        The relative directory path in the space under which the Galaxy data
```
at which

lopiola replied:
może dopisać "If empty, the data will be stored in the space's root directory."

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,30 @@
+  secrets:
+    access_token:
+      label: REST Access Token
+      help: Your access token, suitable for REST API access in a Oneprovider service.
```
może dopisać "May be confined to read-only data access."

lopiola replied:
Można jeszcze rozważyć do tych helpów access token (2x) podlinkowanie https://onedata.org/#/home/documentation/topic/stable/tokens (ale nie wiem jak to będzie wyglądać)

lopiola commented on lib/galaxy/objectstore/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,46 @@
+  secrets:
+    access_token:
+      label: REST Access Token
+      help: Your access token, suitable for REST API access in a Oneprovider service.
```
Może dopisać "Must allow both read and write data access."

nsoranzo committed:
Install node from setup-node action

lopiola commented:
przyjrzec sie opcji writable

lopiola commented:
lopiola: zaproponowac docsy

mvdbeek committed:
Drop extra quote

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

mvdbeek committed:
Merge pull request #18435 from nsoranzo/release_24.0_openapi_setup-node

[24.0] Install node from setup-node action

mvdbeek committed:
Merge pull request #18430 from dannon/fix-chargs-24.0

[24.0] Fix mounting utility usage of pinia (fixes charts form components)

nsoranzo committed:
Split contourpy requirement to support NumPy 2.0

bwalkowi committed:
Unify onedata object store yml and xml configs

bwalkowi committed:
Improve onedata templates docs

bwalkowi committed:
Add writable to onedata file source template

TomHarrop committed:
remove try except block

(cherry picked from commit 5279f446a5bee13a6669d9f3f1a17d11a48cffce)

lopiola commented on lib/galaxy/config/sample/object_store_conf.sample.yml:
```diff
@@ -231,10 +231,13 @@ connection:
   disable_tls_certificate_validation: false
 space:
   # the name of the Onedata space where the Galaxy data will be stored.
```
pierdółka ale te komentarze które są zakończone kropką można by zaczynać duża literą

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,41 @@
+      help: |
+        Your [access token](https://onedata.org/#/home/documentation/topic/stable/tokens),
+        suitable for REST API access in a Oneprovider service. May be confined
+        to read-only data access.
```
May be confined to read-only data access, unless you wish to use the "remote export" tool (write permissions are needed then).

(jeśli ten tool się nie nazywa "remote export" to trezba zaktualizować)

lopiola commented on lib/galaxy/objectstore/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,54 @@
+  name: Onedata Storage
+  version: 0
+  description: |
+    This template allows using Onedata storage.
```
Proponuję

If you have an account in Onedata, you can use such an object store as a storage location for your Galaxy datasets; they will be stored in the Onedata space of your choice. The minimal supported Onezone version is 21.02.4.

bwalkowi replied:
to More on z linkiem zostawić?

lopiola replied:
tak, to zostaje, tylko wymieniamy pierwsze zdanie

lopiola replied:
EDIT zaktualizowałem o wymaganie wersji zona

sveinugu committed:
Added arguments to support loopback to host, which stopped working by default in Podman 5.0

bwalkowi committed:
Update onedata libraries to 21.2.5.2

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,41 @@
+      type: boolean
+      help: |
+        Allow Galaxy to write data to this Onedata storage.
+        Requires access token with write data access.
```
an access token

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,41 @@
+        Your [access token](https://onedata.org/#/home/documentation/topic/stable/tokens),
+        suitable for REST API access in a Oneprovider service. May be confined
+        to read-only data access.
+  configuration:
```
zresolwowałeś comment powyżej ale nie widzę zaktualizowanego tekstu?

lopiola commented on lib/galaxy/config/sample/file_sources_conf.yml.sample:
```diff
@@ -235,6 +235,6 @@
-  onezoneDomain: ${user.preferences['onedata|onezone_domain']}
-  disableTlsCertificateValidation: ${user.preferences['onedata|disable_tls_certificate_validation']}
+  access_token: ${user.preferences['onedata|access_token']}
+  onezone_domain: ${user.preferences['onedata|onezone_domain']}
```
może przy tych preferencjach usera byśmy dopisali "The minimal supported Onezone version is 21.02.4."

lopiola commented on lib/galaxy/config/sample/object_store_conf.sample.yml:
```diff
@@ -220,21 +220,24 @@ extra_dirs:
 connection:
-  # the domain of the Onezone service (e.g. datahub.egi.eu), or its IP address for
+  # The domain of the Onezone service (e.g. datahub.egi.eu), or its IP address for
   # devel instances (see above).
```
dorzuciłbym zdanie "The minimal supported Onezone version is 21.02.4."

bwalkowi committed:
Add info about minimal supported Onezone version

mvdbeek committed:
Fix dropped when_expression on workflow/tool_upgrade

Fixes https://github.com/galaxyproject/galaxy/issues/18441

mvdbeek committed:
Add test to make sure when_expression is not lost on refactoring

mvdbeek committed:
Decrease log level for requesting invalid items

mvdbeek committed:
Raise serializable exceptions when uploading broken workflows

mvdbeek committed:
Raise serializable exception when loading invalid version for a workflow

Fixes https://github.com/galaxyproject/galaxy/issues/18438

martenson committed:
Merge pull request #18171 from dannon/update-security

[PROCEDURES] Minor adjustments to SECURITY.md to reflect that we call yearly releases LTS

mvdbeek committed:
Merge pull request #18428 from mvdbeek/replace_busybox

[24.0] Replace busybox:ubuntu-14.04 image with busybox:1.36.1-glibc

dannon committed:
Merge remote-tracking branch 'upstream/release_24.0' into release_24.1

dannon committed:
Merge remote-tracking branch 'upstream/release_24.1' into dev

mvdbeek committed:
Merge pull request #18447 from mvdbeek/exception_cleanup

[24.1] Improve workflow-related exception reporting

bernt-matthias committed:
Minor additions to release process

martenson committed:
adjust committers membership

guerler committed:
Remove redundant home icon from masthead

bernt-matthias committed:
Create version 24.0.3

bernt-matthias committed:
Start work on 24.0.4.dev0

bernt-matthias committed:
Merge branch 'release_24.0' into release_24.1

# Conflicts:
#	lib/galaxy/version.py
#	packages/app/HISTORY.rst
#	packages/app/setup.cfg
#	packages/auth/HISTORY.rst
#	packages/auth/setup.cfg
#	packages/config/HISTORY.rst
#	packages/config/setup.cfg
#	packages/data/HISTORY.rst
#	packages/data/setup.cfg
#	packages/files/HISTORY.rst
#	packages/files/setup.cfg
#	packages/job_execution/HISTORY.rst
#	packages/job_execution/setup.cfg
#	packages/job_metrics/HISTORY.rst
#	packages/job_metrics/setup.cfg
#	packages/navigation/HISTORY.rst
#	packages/navigation/setup.cfg
#	packages/objectstore/HISTORY.rst
#	packages/objectstore/setup.cfg
#	packages/schema/HISTORY.rst
#	packages/schema/setup.cfg
#	packages/selenium/HISTORY.rst
#	packages/selenium/setup.cfg
#	packages/test_api/HISTORY.rst
#	packages/test_api/setup.cfg
#	packages/test_base/HISTORY.rst
#	packages/test_base/setup.cfg
#	packages/test_driver/HISTORY.rst
#	packages/test_driver/setup.cfg
#	packages/tool_shed/HISTORY.rst
#	packages/tool_shed/setup.cfg
#	packages/tool_util/HISTORY.rst
#	packages/tool_util/setup.cfg
#	packages/tours/HISTORY.rst
#	packages/tours/setup.cfg
#	packages/util/HISTORY.rst
#	packages/util/setup.cfg
#	packages/web_apps/HISTORY.rst
#	packages/web_apps/setup.cfg
#	packages/web_framework/HISTORY.rst
#	packages/web_framework/setup.cfg
#	packages/web_stack/HISTORY.rst
#	packages/web_stack/setup.cfg

bernt-matthias committed:
Merge branch 'release_24.1' into dev

arash77 committed:
Refactor WorkflowContentsManager to use a list for tags instead of a string

bernt-matthias committed:
Update doc/source/dev/create_release.rst

Co-authored-by: Martin Cech <marten@bx.psu.edu>

jdavcs committed:
Update doc/source/dev/create_release.rst

jmchilton committed:
Stray small changes from structured_tool_state branch.

guerler committed:
Adjust masthead test case

jdavcs committed:
Merge pull request #18450 from galaxyproject/bernt-matthias-patch-1

Minor additions to release process

guerler committed:
Disable storage dashboard for anon users

guerler committed:
Link to login instead of disabling storage tab

guerler committed:
Add prevent directive to masthead item click handler

arash77 committed:
Add test

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,42 @@
+  description: |
+    If you have an account in Onedata, you can use such a file source to upload
+    your data directly from Onedata to Galaxy. You can also perform
+    a "Export datasets to remote files source" of a dataset from Galaxy to
```
an Export

lopiola commented on lib/galaxy/files/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,42 @@
+    If you have an account in Onedata, you can use such a file source to upload
+    your data directly from Onedata to Galaxy. You can also perform
+    a "Export datasets to remote files source" of a dataset from Galaxy to
+    Onedata. The minimal supported Onezone version is 21.02.4.
```
chyba lepiej by było dać odstęp newline przed "The minimal"

lopiola commented on lib/galaxy/objectstore/templates/examples/production_onedata.yml:
```diff
@@ -0,0 +1,54 @@
+  description: |
+    If you have an account in Onedata, you can use such an object store as
+    a storage location for your Galaxy datasets; they will be stored in
+    the Onedata space of your choice. The minimal supported Onezone version
```
podobnie, proponuję odstęp przed The minimal

sanjaysrikakulam committed:
update plausible script name in mako templates

sanjaysrikakulam committed:
update plausible script name in mako templates

laperlej committed:
Prevent deleted users from reseting password

guerler committed:
Limit username text length in masthead

guerler committed:
Update About Page?

mvdbeek committed:
Add input extra files to `get_input_fnames`

This should fix volume mounts in kubernetes,
which calls:

```
volume_mounts.extend(generate_relative_mounts(data_claim, job_wrapper.job_io.get_input_fnames()))
```
The only other consumers are pbs, using an undocument config option
(ahem ...).

dannon committed:
Move workflow preview to primary link for workflows card list -- easier to find and feels more natural

guerler committed:
Improve brand alignment padding

dannon committed:
Trim whitespace after link to avoid an underlined trailing space

itisAliRH committed:
🛠️: dropping for loop in `WorkflowActions` to improve readability, use button group to fix ui issues and remove AsyncButton usage

itisAliRH committed:
🛠️: remove unused props in `WorkflowActionsExtend`

guerler committed:
Rephrase some labels, and add v-localize directives

itisAliRH committed:
🚚:  move `WorkflowList` related components from `components/Workflow` to `components/Workflow/List`

laperlej committed:
Update lib/galaxy/managers/users.py

Co-authored-by: John Davis <jdavcs@gmail.com>

laperlej committed:
fix indentation

guerler committed:
Remove data libraries from masthead

guerler committed:
Add data libraries to activities

guerler committed:
Merge pull request #17927 from guerler/masthead_update

Masthead Revision

guerler committed:
Move libraries to activities

laperlej committed:
combine user exist check with user deleted check

guerler committed:
Remove data library from galaxy ui tour

martenson committed:
Merge pull request #18459 from laperlej/reset_password_deleted_user

[24.0] Disable password reset for deleted users [GCC2024_COFEST]

mvdbeek committed:
Merge pull request #18456 from arash77/subworkflows_tags_fix

[24.0] Fix subwofklow tags serialization type

guerler committed:
Restore h1 header for about page

guerler committed:
Adjust masthead jest test

guerler committed:
Merge pull request #18461 from guerler/about_page_update

Updates the About Page

mvdbeek committed:
Merge pull request #18462 from mvdbeek/add_input_extra_files_path

[24.0] Add input extra files to `get_input_fnames`

martenson committed:
Merge pull request #18458 from sanjaysrikakulam/plausible_script

update plausible script name in mako templates

martenson committed:
Merge pull request #18463 from dannon/workflow-preview-link

Move workflow preview to primary link for workflows card list -- easi…

nsoranzo committed:
Fix new E721 and F811 errors from ruff 0.5.0

nsoranzo committed:
Install isa-rwval from my fork to relax pin on networkx

xref. https://github.com/ISA-tools/isa-rwval/pull/17

natefoo committed:
Make `default_panel_view` a `_by_host` option

jmchilton committed:
Fix unit test.

bernt-matthias committed:
22.05 pin bleach to 5.0.1

22.05 is currently failing planemo tests because bleach 5.0.0
has an error in its requirements that has been fixed here
https://github.com/mozilla/bleach/pull/655

ElectronicBlueberry committed:
detect markdown changes before notifying

mvdbeek committed:
Merge pull request #18472 from bernt-matthias/22.05-bleach5.01

[22.05] pin bleach to 5.0.1

mvdbeek committed:
Merge pull request #18446 from mvdbeek/when_upgrade_fix

[24.0] Fix dropped when_expression on step upgrade

mvdbeek committed:
Merge branch 'release_22.05' into release_23.0

mvdbeek committed:
Merge branch 'release_23.0' into release_23.1

mvdbeek committed:
Merge branch 'release_23.1' into release_23.2

mvdbeek committed:
Merge pull request #18474 from martenson/backport-18458

[24.0] Backport 18458

mvdbeek committed:
Merge branch 'release_23.2' into release_24.0

mvdbeek committed:
Merge branch 'release_24.0' into release_24.1

mvdbeek committed:
Merge branch 'release_24.1' into dev

mvdbeek committed:
Merge pull request #18433 from nsoranzo/fix_deprecations

Fix some deprecations

ElectronicBlueberry committed:
refactor out repeated functionality

ahmedhamidawan committed:
[24.0] Return generic message for password reset email

This prevents existence of a user account from being queryable through password reset. We now return `None` and display a generic message regardless of a prt being created or not.

Fixes https://github.com/galaxyproject/galaxy/issues/18475

jdavcs committed:
Merge pull request #18439 from sveinugu/podman5_fix

Fixed Podman 5 issue in sample job config file

ahmedhamidawan committed:
adjust `test_reset_email_user_deleted` for `None` instead of str

ahmedhamidawan committed:
remove else block and return `None` by the outer conditional

ahmedhamidawan committed:
Update test/unit/app/managers/test_UserManager.py

Co-authored-by: John Davis <jdavcs@gmail.com>

itisAliRH committed:
✔️: add workflow preview class and make selenium test pass

dannon committed:
Merge pull request #18483 from itisAliRH/fix-workflow-preview-selenium-test

Fix Workflow Preview Selenium Test Error

dannon committed:
Merge pull request #18465 from itisAliRH/workflow-list-refactors

Workflow Card Buttons Refactors [GCC2024_COFEST]

itisAliRH committed:
✔️: fix WorkflowFilters path in filterConversion unit test

mvdbeek committed:
Merge pull request #18453 from jmchilton/structured_tool_state_small_things

Stray small changes from structured_tool_state branch.

jdavcs committed:
Create version 24.1.1

jdavcs committed:
Start work on 24.1.2.dev0

jdavcs committed:
Merge branch 'release_24.1' into dev

# Conflicts:
#	lib/galaxy/version.py
#	packages/app/HISTORY.rst
#	packages/auth/HISTORY.rst
#	packages/config/HISTORY.rst
#	packages/data/HISTORY.rst
#	packages/files/HISTORY.rst
#	packages/job_execution/HISTORY.rst
#	packages/job_metrics/HISTORY.rst
#	packages/navigation/HISTORY.rst
#	packages/objectstore/HISTORY.rst
#	packages/schema/HISTORY.rst
#	packages/selenium/HISTORY.rst
#	packages/test_api/HISTORY.rst
#	packages/test_base/HISTORY.rst
#	packages/test_driver/HISTORY.rst
#	packages/tool_shed/HISTORY.rst
#	packages/tool_util/HISTORY.rst
#	packages/tours/HISTORY.rst
#	packages/util/HISTORY.rst
#	packages/web_apps/HISTORY.rst
#	packages/web_framework/HISTORY.rst
#	packages/web_stack/HISTORY.rst

jdavcs committed:
Update package dev versions

guerler committed:
Merge pull request #18466 from itisAliRH/workflow-list-move-files

Move Workflow List Files to New Location [GCC2024_COFEST]

mvdbeek committed:
Merge pull request #18473 from ElectronicBlueberry/fix-workflow-relabel-noop

[24.1] fix no-op workflow "relabel" shouldn't result in a notification

mvdbeek committed:
Merge pull request #18471 from natefoo/default-panel-view-by-host

[24.1] Make `default_panel_view` a `_by_host` option

mvdbeek committed:
Merge pull request #18489 from jdavcs/dev_package_dev_versions

Update package dev versions

mvdbeek committed:
Merge branch 'release_24.1' into dev

davelopez committed:
Add TypeScript support to Invocation Export related files

nsoranzo committed:
Fix unit test

martenson committed:
Merge pull request #18468 from guerler/move_libraries

Moves Libraries from Masthead to Activity Bar

mvdbeek committed:
Raise ValueError if invalid ID passed

I suspect a state bug in the tool form code is the issue here and the id
refers to a HDA id, but this is a solid fix anyway if someone submits
wrong values via the API.
Fixes https://github.com/galaxyproject/galaxy/issues/18485:
```
Stack Trace

Newest

AssertionError: null
  File "galaxy/web/framework/decorators.py", line 346, in decorator
    rval = func(self, trans, *args, **kwargs)
  File "galaxy/webapps/galaxy/api/tools.py", line 247, in build
    return tool.to_json(trans, kwd.get("inputs", kwd), history=history)
  File "galaxy/tools/__init__.py", line 2510, in to_json
    populate_state(request_context, self.inputs, params.__dict__, state_inputs, state_errors)
  File "galaxy/tools/parameters/__init__.py", line 412, in populate_state
    _populate_state_legacy(
  File "galaxy/tools/parameters/__init__.py", line 625, in _populate_state_legacy
    check_param(request_context, input, param_value, context, simple_errors=simple_errors)
  File "galaxy/tools/parameters/__init__.py", line 246, in check_param
    value = param.from_json(value, trans, param_values)
  File "galaxy/tools/parameters/basic.py", line 2132, in from_json
    value = self.to_python(value, trans.app)
  File "galaxy/tools/parameters/basic.py", line 1963, in to_python
    return history_item_dict_to_python(value["values"][0], app, self.name)
  File "galaxy/tools/parameters/basic.py", line 2799, in history_item_dict_to_python
    return src_id_to_item(sa_session=app.model.context, security=app.security, value=value)
  File "galaxy/tools/parameters/basic.py", line 2056, in src_id_to_item
    assert item
```

mvdbeek committed:
Downgrade missing workflow outputs to warning

Fixes https://sentry.galaxyproject.org/share/issue/3fa02a539eee4787a465c0b99903df3a/

mvdbeek committed:
Use get_file_obj to render compressed data in mako

We were mostly doing this already. Also raises an exception in
`text_data` if content is not decodable.

Fixes
https://sentry.galaxyproject.org/share/issue/b826c0be95a4424e8b1d63983255f23d/:
```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
  File "galaxy/web/framework/middleware/error.py", line 167, in __call__
    app_iter = self.application(environ, sr_checker)
  File "galaxy/web/framework/middleware/statsd.py", line 29, in __call__
    req = self.application(environ, start_response)
  File "/cvmfs/main.galaxyproject.org/venv/lib/python3.11/site-packages/paste/httpexceptions.py", line 635, in __call__
    return self.application(environ, start_response)
  File "galaxy/web/framework/base.py", line 174, in __call__
    return self.handle_request(request_id, path_info, environ, start_response)
  File "galaxy/web/framework/base.py", line 263, in handle_request
    body = method(trans, **kwargs)
  File "galaxy/webapps/galaxy/controllers/dataset.py", line 152, in display
    display_data, headers = data.datatype.display_data(
  File "galaxy/datatypes/tabular.py", line 199, in display_data
    truncated_data=open(dataset.get_file_name()).read(max_peek_size),
  File "<frozen codecs>", line 322, in decode
```

mvdbeek committed:
Fix statsd client on urls that aren't ascii encodable

Fixes https://sentry.galaxyproject.org/share/issue/e2c535d2d4904d438a0a1524c991bba4/:
```
UnicodeEncodeError: 'ascii' codec can't encode characters in position 14-16: ordinal not in range(128)
  File "galaxy/web/framework/middleware/error.py", line 167, in __call__
    app_iter = self.application(environ, sr_checker)
  File "galaxy/web/framework/middleware/statsd.py", line 34, in __call__
    self.galaxy_stasd_client.timing(page, dt)
  File "galaxy/web/statsd_client.py", line 35, in timing
    self.statsd_client.timing(infix + path, time)
  File "statsd/client/base.py", line 33, in timing
    self._send_stat(stat, '%0.6f|ms' % delta, rate)
  File "statsd/client/base.py", line 61, in _send_stat
    self._after(self._prepare(stat, value, rate))
  File "statsd/client/base.py", line 76, in _after
    self._send(data)
  File "statsd/client/udp.py", line 42, in _send
    self._sock.sendto(data.encode('ascii'), self._addr)
```

davelopez committed:
Convert ExportToRemoteButton to Typescript

mvdbeek committed:
Merge pull request #18406 from galaxybot/dev_auto_update_dependencies

Update Python dependencies

sanjaysrikakulam committed:
add missing version in the file sources and object store templates

davelopez committed:
Expose status property in useTaskMonitor

davelopez committed:
Add short-term storage monitor composable and tests

davelopez committed:
Update short-term storage poll frequency to 3 seconds

davelopez committed:
refactor: Simplify short-term storage composable and reuse monitor

davelopez committed:
Add generic task monitor composable

davelopez committed:
Refactor task monitor to use generic composable

davelopez committed:
Refactor short-term storage monitor to use generic composable

mvdbeek committed:
Stabilize test_purge_while_job_running test

In the extended_metadata + remote tool_evaluation_strategy the
commandline is templated as part of the job, so we need to give the test
more time to start running before we delete outputs (as then we can't
template the command line anymore).

davelopez committed:
Add TaskMonitor interface for generic task monitor

davelopez committed:
Add persistentProgressMonitor composable

It allows to keep track of the progress of a long-running operation
even if the user navigates away from the page.

mvdbeek committed:
Merge pull request #18498 from mvdbeek/stabilize_test_purge_while_job_running

[24.1] Stabilize test_purge_while_job_running test

mvdbeek committed:
Merge pull request #18494 from mvdbeek/fix_invalid_id_error_message

[24.1] Increase API robustness to invalid requests, improve compressed data serving

mvdbeek committed:
Raise Message Exception when displaying binary data

Fixes
https://sentry.galaxyproject.org/share/issue/a8843884527f4e4089b32fd14a2f126d/:
```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 4: invalid start byte
  File "galaxy/web/framework/middleware/error.py", line 167, in __call__
    app_iter = self.application(environ, sr_checker)
  File "galaxy/web/framework/middleware/statsd.py", line 29, in __call__
    req = self.application(environ, start_response)
  File "/cvmfs/main.galaxyproject.org/venv/lib/python3.11/site-packages/paste/httpexceptions.py", line 635, in __call__
    return self.application(environ, start_response)
  File "galaxy/web/framework/base.py", line 174, in __call__
    return self.handle_request(request_id, path_info, environ, start_response)
  File "galaxy/web/framework/base.py", line 263, in handle_request
    body = method(trans, **kwargs)
  File "galaxy/webapps/galaxy/controllers/dataset.py", line 152, in display
    display_data, headers = data.datatype.display_data(
  File "galaxy/datatypes/sequence.py", line 785, in display_data
    "/dataset/large_file.mako", truncated_data=fh.read(max_peek_size), data=dataset
  File "<frozen codecs>", line 322, in decode

```
Which is a BAM file assigned to fastqsanger.gz

ahmedhamidawan committed:
Update lib/galaxy/managers/users.py

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

davelopez committed:
Add status and stored task ID to persistent progress monitor

davelopez committed:
Add unit tests for usePersistentProgressTaskMonitor

martenson committed:
Merge pull request #18479 from ahmedhamidawan/prevent_user_account_queryable_on_reset

[24.0] Return generic message for password reset email

bgruening committed:
Merge pull request #18503 from mvdbeek/error_on_binary_data_display

Raise Message Exception when displaying binary data

martenson committed:
Merge pull request #18507 from TomHarrop/datatype_pod5_backport

[24.1] Backport pod5 datatype

nsoranzo committed:
Handle all requests error in ``ApiBiotoolsMetadataSource._raw_get_metadata``

Fix the following traceback when running `planemo lint` if bio.tools is down:

```
File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/galaxy/tool\_util/biotools/source.py", line 69, in get\_biotools\_metadata
return BiotoolsEntry.from\_json(json.loads(content))
^^^^^^^^^^^^^^^^^^^
File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/json/**init**.py", line 346, in loads
return \_default\_decoder.decode(s)
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/json/decoder.py", line 337, in decode
obj, end = self.raw\_decode(s, idx=\_w(s, 0).end())
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/json/decoder.py", line 355, in raw\_decode
raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

davelopez committed:
Add some docs to TaskMonitor interface

davelopez committed:
Add PersistentTaskProgressMonitorAlert component

The PersistentTaskProgressMonitorAlert component is added to provide a visual representation of the progress of a long-running operation. It includes different alert messages for different states of the task, such as in progress, completed, failed, and request failed. The component also supports automatic download of the task result when it is completed, if enabled.

davelopez committed:
Replace export logic in InvocationExportPluginCard

This will improve the export experience when the process is long so the
user can navigate away from the page and still see the progress of the
export or be able to download the result when it's ready.

- Remove the ExportToRemoteButton and the StsDownloadButton and replace them with a simpler ExportButton that only reacts to the busy state.
- Add 2 persistent progress trackers (alert) one for each type of export task.

davelopez committed:
Drop unnecessary test

This is now tested in the PersistentTaskProgressMonitorAlert component

nsoranzo committed:
Label issues with ``area/UI-UX`` only when both criteria match

mvdbeek committed:
Add test for submitting hdca with deleted elements

nsoranzo committed:
Adding the milestone to a PR requires ``issues: write`` permission

in addition to ``pull-requests: write``, see
https://github.com/actions/github-script/issues/255 .

nsoranzo committed:
Merge pull request #18510 from nsoranzo/release_23.2_fix_ApiBiotoolsMetadataSource._raw_get_metadata

[23.2] Handle all requests error in ``ApiBiotoolsMetadataSource._raw_get_metadata``

nsoranzo committed:
Merge branch 'release_23.2' into release_24.0

nsoranzo committed:
Merge branch 'release_24.0' into release_24.1

nsoranzo committed:
Merge branch 'release_24.1' into dev

jdavcs committed:
Merge pull request #18516 from nsoranzo/release_24.0_followup_on_18212

[24.0] Follow-up on #18212

nsoranzo committed:
Merge branch 'release_24.0' into release_24.1

nsoranzo committed:
Merge branch 'release_24.1' into dev

nsoranzo committed:
Merge pull request #18452 from martenson/com-proc

[PROCEDURES] remove davebx from committers list

davelopez committed:
Update openapi-typescript dependency to version 6.7.6

And re-generate schema.ts

davelopez committed:
Fix copy collection API return code + client typing

davelopez committed:
Fix VariableValueType typing to match API type

And adapt formUtil functions to handle undefined values

jmchilton committed:
Merge pull request #18497 from sanjaysrikakulam/update_file_and_objectstore_templates

Add missing version in the file sources and object store templates

mvdbeek committed:
Cache and provide in-memory fallback

dannon committed:
Merge pull request #18512 from davelopez/improve_invocation_export_ui

Improve invocation export UI

davelopez committed:
Update test_jobs.py to add a new test for job list collection view

davelopez committed:
Fix view parameter type in JobsService index

davelopez committed:
Add test for default view in job list

davelopez committed:
Fix default view parameter type in jobs index API endpoint

mvdbeek committed:
Merge pull request #18521 from davelopez/24.0_fix_api_job_view_type

[24.0] Fix view parameter type in Job index API

mvdbeek committed:
Merge pull request #18517 from mvdbeek/fix_deleted_ipnut_submission_collection

[24.1] Prevent job submission if input collection element is deleted

abretaud committed:
Fix worklfow typo here and there

dannon committed:
Merge pull request #18523 from abretaud/worklfow

Fix "worklfow" typo here and there

mvdbeek committed:
Merge pull request #18519 from davelopez/update_openapi_typescript_6.7.6

Update openapi-typescript dependency to version 6.7.6

mvdbeek committed:
Merge branch 'release_24.0' into release_24.1

mvdbeek committed:
Merge branch 'release_24.1' into dev

lopiola committed:
Merge branch 'dev' of github.com:galaxyproject/galaxy into add-onedata-templates

# Conflicts:
#	lib/galaxy/dependencies/dev-requirements.txt

bwalkowi committed:
Remove production_ prefix from onedata templates
